app/controllers/samples_controller.rb

```diff
@@ -234,14 +234,34 @@ def report_info
     @report_info = external_report_info(pipeline_run_id, background_id, params)
 
     tax_ids = @report_info[:taxonomy_details][2].map { |x| x['tax_id'] }
+    tax_ids = tax_ids | @report_info[:taxonomy_details][2].map { |x| x['family_taxid'] }
     lineages = TaxonCount.connection.select_all(TaxonLineage.where(taxid: tax_ids)).to_hash
     lineage_by_taxid = {}
     lineages.each do |x|
       lineage_by_taxid[x['taxid']] = x
     end
 
     @report_info[:taxonomy_details][2].each do |tax|
-      tax['lineage'] = lineage_by_taxid[tax['tax_id']]
+      if tax['tax_id'] > 0
```

Comment:
can you make 245-264 a function that can be changed around later? 
also can extract another function to extract the level of the match. But maybe there's an easier way for representation in TaxonCount. Also more comments for this function would be appreciated.